src/main/MainTabsScreen.js

```diff
@@ -56,25 +72,19 @@ export default function MainTabsScreen(props: Props): Node {
         <Tab.Screen
           name="home"
           component={HomeScreen}
-          options={{
-            tabBarLabel: 'Home',
-            tabBarIcon: ({ color }) => <IconInbox size={24} color={color} />,
-          }}
+          options={{ tabBarLabel: 'Home', tabBarIcon: HomeScreenIcon }}
```

Comment:
Hmm, it seems like a regression in clarity to have to separate like this the definition of what icon we use from the rest of the description of this tab -- the text label, and what screen component it is.

What if you use `useMemo`, like I suggested above?

… I see, the lint rule doesn't accept it:
```js
          options={React.useMemo(
            () => ({
              tabBarLabel: 'Home',
              tabBarIcon: ({ color }) => <IconInbox size={24} color={color} />,
            }),
            [],
          )}
```

>  62:27  error  Declare this component outside parent component "MainTabsScreen" or memoize it. If you want to allow component creation in props, set allowAsProps option to true  react/no-unstable-nested-components

So even though the error message says "memoize it", if it's wrapped in a `React.useMemo` right there, it still complains. That's pretty buggy behavior for that lint rule.

(Same result if I spell it just `useMemo`, after importing it directly.)

Ahh, consulting the docs:
https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-unstable-nested-components.md
they have an example mentioning `React.useCallback`. So that's what they seem to have in mind, and they just forgot about `useMemo`.

… But that doesn't work either:
```js
            tabBarIcon: React.useCallback(({ color }) => <IconInbox size={24} color={color} />, []),
```
That still gets the same error.

Oho, and the docs also remind me (while discussing the rule's `allowAsProps` option) that having a fresh value every time is fine if what's done with the value is to *call* it within a render function, and not to use it as a React element type:
https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-unstable-nested-components.md#rule-options

Digging into the react-navigation implementation, that is in fact what's done with this `tabBarIcon` option. The name gets changed a couple of times as it gets passed through layers, but here's where it gets consumed, in `node_modules/@react-navigation/bottom-tabs/src/views/TabBarIcon.tsx` -- it's called `renderIcon` here:
```jsx
      <View style={[styles.icon, { opacity: activeOpacity }]}>
        {renderIcon({
          focused: true,
          size,
          color: activeTintColor,
        })}
      </View>
```

I think that means we should use that `allowAsProps` option, and leave this code as it is.